prisma/schema.prisma

```diff
@@ -11,14 +11,111 @@ datasource db {
+  pilots  Pilot[]
 }
 
-enum Level {
-  Info
-  Warn
-  Error
+model ShipAbility {
+  id    String  @id @default(uuid())
+  name  String
+  text  String
+  Pilot Pilot[]
+}
+
+model Pilot {
+  id             String        @id @default(uuid())
+  faction        String
+  shipId         String
+  ship           Ship          @relation(fields: [shipId], references: [id])
+  name           String
+  subtitle       String
+  initiative     Int
+  force          Int // XWS has this as a bag with the force and forceRecover packed... I don't have a gut sense of what I like better but split out for stronger typing for now
+  forceRecover   Int
+  charges        Int // Same as force, packed in XWS
+  chargesRecover Int
+  limit          Int // Fighting my inner pedant here - more technically correct feels like a "limited" boolean and then a nullable "limit" int, but a limit of 0 accomplishes the same
+  pilotAbility   String
+  shipAbilityId  String
+  shipAbility    ShipAbility   @relation(fields: [shipAbilityId], references: [id])
+  PilotPoints    PilotPoints[]
+
+  @@unique([faction, name, subtitle])
+}
+
+model Upgrade {
+  id    String        @id @default(uuid())
+  name  String
+  types UpgradeType[]
+  text  String
+
+  @@unique([name, types])
+  PilotPoints   PilotPoints?    @relation(fields: [pilotPointsId], references: [id])
+  pilotPointsId String?
+  UpgradePoints UpgradePoints[]
+}
+
+model PointsEnvironment {
+  id             String          @id @default(uuid())
+  releaseDate    DateTime?
+  effectiveDate  DateTime?
+  source         String
+  version        String?
+  pilotsPoints   PilotPoints[]
+  upgradesPoints UpgradePoints[]
+}
+
+model PilotPoints {
+  id                  String            @id @default(uuid())
+  pointsEnvironmentId String
+  pointsEnvironment   PointsEnvironment @relation(fields: [pointsEnvironmentId], references: [id])
+  pilotId             String
+  pilot               Pilot             @relation(fields: [pilotId], references: [id])
+  cost                Int
+  loadoutValue        Int
+  upgradeBar          Upgrade[]
+  keywords            Keyword[]
+  inStandard          Boolean
+  inExtended          Boolean
+
+  @@unique([pointsEnvironmentId, pilotId])
+}
+
+model UpgradePoints {
+  pointsEnvironmentId String
+  pointsEnvironment   PointsEnvironment @relation(fields: [pointsEnvironmentId], references: [id])
+  upgradeId           String
+  upgrade             Upgrade           @relation(fields: [upgradeId], references: [id])
+  cost                Int
+  restrictions        Restriction[]
+  inStandard          Boolean
+  inExtended          Boolean
+
+  @@unique([pointsEnvironmentId, upgradeId])
+}
+
+//
+// This feels annoying and fiddly, but better than typing "Imperial" everywhere and not catching the time you accidentally
+// typed "Impreial"
+//
+enum Keyword {
+  Imperial
+}
+
+//
+// Wait - are Restrictions and Keywords the same set? Yes?
```

Comment:
I believe so, yes - but, likely makes sense to have different models for pilots and upgrades.
For pilots, factions are a restriction for what the pilot can be played in. Keywords are a map of additional benefits, effectively.

For upgrades, factions and keywords are both restrictions. I don't think that means the right answer is to make a second set of effectively identical enums, but likely the naming conventions for how these are stored on pilots/upgrades?

For example, all pilots must have exactly 1 faction. An Upgrade has a list of requirements, some of which may be factions.

Reply:
Okay - looks like XWS is doing this by bagging the different requirements, e.g.

```
"restrictions": [{ "keywords": ["Mandalorian"] }],
```

I think perhaps the strongly-typed way of doing this is to split the different types of restrictions?

```
factionRestrictions
keywordRestrictions
baseSizeRestrictions
```

Actually - I need to confirm if Prisma can do arrays of relative models. I assume so, but don't actually know that for sure. Assuming it can, I think this feels cleanest to me.